### PR TITLE
2331 refactor/rename column mapping profile

### PR DESCRIPTION
**Pull request by macintoshpie**
#### What's this PR do? 
Renames the ColumnMappingPreset model to ColumnMappingProfile. Updates all names (frontend and backend) from preset to profile as well.
Some files were not fully renamed, specifically any of the url/view files related to v2 and v2.1 apis.

Note, however this change _does_ require that use of these older endpoints now provide payloads that correspond to the new model (specifically, the attribute `preset_type` was changed to `profile_type`).

This is most likely not a commonly used api (and a relatively new field as well, within the last 3 months) so this is probably not a large issue, just something to be aware of.

#### How should this be manually tested?
Apply the migrations to update the schemas. Hard refresh your browser page to reset the cache

Column mapping profiles are used in a couple of places:
- `/app/#/accounts/<org id>/column_mappings/properties` (organization column mappings tab).
  - Verify "presets" isn't seen anywhere on screen.
  - You should see all of the orgs profiles in the dropdown
  - try: creating, updating, changing the name, and deleting profiles
  - see buildingsync test notes below
- `/app/#/properties/<id>` (property details page)
  - click the "Export BuildlingSync" action. The modal should show all of your orgs buildingsync profiles.
  - see the bsync test notes below
- `/app/#/data/mapping/5970` (data mapping page)
  - you should see the profiles populate the dropdown (according to the type of file you're importing)
  - create a new preset
  - see buildingsync test notes below

Also updated the service and view for downloading buildingsync. Test by going to inventory detail page then choosing the buildingsync export action.

There are some BuildingSync specific tests you should do as well. See the manual testing section in this PR: https://github.com/SEED-platform/seed/pull/2213

#### What are the relevant tickets?
#2331 
#2367

### CODE REVIEW

File: seed/static/seed/js/controllers/mapping_controller.js [hunk 1]
```diff
@@ -64,112 +64,123 @@ angular.module('BE.seed.controller.mapping', [])
...
-        return $scope.current_preset.id === 0
-          || $scope.preset_change_possible
+        // BuildingSync requires a saved profile to be applied
+        return $scope.current_profile.id === 0
+          || $scope.profile_change_possible
           || $scope.mappings_change_possible;
       };
 
-      $scope.apply_preset = function () {
+      var analyze_chosen_inventory_types = function () {
```
Comment (adrian-lara) [inline seed/static/seed/js/controllers/mapping_controller.js:100]: I think I did this... I'm a bit worried now that I botched the merge conflict using the GH UI to do it 🤦 .

This function was removed here: https://github.com/SEED-platform/seed/pull/2372/files

Thinking we revert locally, update with develop, then force push up?

File: seed/views/v3/properties.py [hunk 2]
```diff
@@ -1193,22 +1193,22 @@ def building_sync(self, request, pk):
         """
         Return BuildingSync representation of the property
         """
-        preset_pk = request.GET.get('preset_id')
```
Comment (adrian-lara) [inline seed/views/v3/properties.py:1196]: Still some hints of "preset" in this file that I think we can change now.

File: seed/migrations/0128_auto_20200810_1731.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('orgs', '0013_organization_comstock_enabled'),
+        ('seed', '0127_auto_20200605_1125'),
+    ]
+
+    operations = [
+        migrations.RenameModel(
```
[the remark this answers is not in the document]
Comment (macintoshpie) [reply seed/migrations/0128_auto_20200810_1731.py:14]: Yup! It's a 2.0 feature
https://stackoverflow.com/questions/25091130/django-migration-strategy-for-renaming-a-model-and-relationship-fields/26241640?noredirect=1#comment86571893_26241640

File: seed/migrations/0129_auto_20200810_2020.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('seed', '0128_auto_20200810_1731'),
+    ]
+
+    operations = [
+        migrations.RenameField(
```
[the remark this answers is not in the document]
Comment (macintoshpie) [reply seed/migrations/0129_auto_20200810_2020.py:13]: Yeah I can do that, I did it separately to make sure django could pickup the changes, but there's no reason we can't add them together 👍

File: seed/migrations/0126_columnmappingpreset_preset_type.py [hunk 5]
```diff
@@ -13,7 +13,7 @@ def create_default_bsync_presets(apps, schema_editor):
         bsync_mapping_name = 'BuildingSync v2.0 Defaults'
         org.columnmappingpreset_set.create(
             name=bsync_mapping_name,
-            mappings=default_buildingsync_preset_mappings(),
+            mappings=default_buildingsync_profile_mappings(),
```
[the remark this answers is not in the document]
Comment (macintoshpie) [reply seed/migrations/0126_columnmappingpreset_preset_type.py:16]: That's correct, it's a deterministic transformation of a 'constant' dictionary

File: seed/views/v3/properties.py (hunk 2 above)
Comment (macintoshpie) [reply seed/views/v3/properties.py:1196]: Ah good find, I'll make that change

File: seed/static/seed/js/controllers/export_buildingsync_modal_controller.js [hunk 6]
```diff
@@ -10,22 +10,29 @@ angular.module('BE.seed.controller.export_buildingsync_modal', [])
...
       $uibModalInstance,
       property_view_id,
-      column_mapping_profiles
+      column_mapping_profiles,
+      user_service,
     ) {
       $scope.column_mapping_profiles = column_mapping_profiles;
       $scope.current_column_mapping_profile = column_mapping_profiles[0];
 
       $scope.download_file = function () {
```
Comment (macintoshpie) [inline seed/static/seed/js/controllers/export_buildingsync_modal_controller.js:26]: Please test this by going to inventory detail page, then selecting export as buildingsync action

File: seed/static/seed/locales/en_US.json [hunk 7]
```diff
@@ -46,11 +46,11 @@
     "Address Line 2 (Tax Lot)": "Address Line 2 (Tax Lot)",
     "Admin": "Admin",
     "An error occurred while processing the file. Please ensure that your file meets the required specifications.": "An error occurred while processing the file. Please ensure that your file meets the required specifications.",
-    "AND": "AND",
     "and": "and",
+    "AND": "AND",
```
Comment (macintoshpie) [inline seed/static/seed/locales/en_US.json:50]: Not sure why this happened for all capitalization alternatives...

File: locale/en_US/LC_MESSAGES/django.po [hunk 8]
```diff
@@ -173,8 +173,8 @@ msgstr "api key"
 msgid "API Key"
 msgstr "API Key"
 
-msgid "Apply Preset"
```
[the remark this answers is not in the document]
Comment (macintoshpie) [reply locale/en_US/LC_MESSAGES/django.po:176]: I updated the keys, native language, and translations on the lokalise site and updated en and fr json files.

To test, I visited the following pages looking for broken translations:
- Organization Column Mappings settings page
- mapping page (load profile, save profile, etc)
- export buidingsync (asks for profile selection)

File: seed/static/seed/locales/en_US.json (hunk 7 above)
Comment (adrian-lara) [reply seed/static/seed/locales/en_US.json:50]: Likely these were changes made prior and not imported into SEED. Should be fine that it changed.

File: locale/en_US/LC_MESSAGES/django.po (hunk 8 above)
[the remark this answers is not in the document]
Comment (adrian-lara) [reply locale/en_US/LC_MESSAGES/django.po:176]: I think I forgot to go over this one any detail (which I absolutely should have 😬 )... Let's have a 5 min chat when you get a chance.

File: seed/static/seed/js/controllers/export_buildingsync_modal_controller.js (hunk 6 above)
Comment (adrian-lara) [reply seed/static/seed/js/controllers/export_buildingsync_modal_controller.js:26]: Works as expected 👍

**Pull request merged**